src/reconcile/rec_visibility.c

```diff
@@ -278,9 +281,33 @@ __rec_need_save_upd(
         return (false);
 
     if (WT_TIME_WINDOW_HAS_STOP(&upd_select->tw))
-        return (!__wt_txn_tw_stop_visible_all(session, &upd_select->tw));
+        visible_all = __wt_txn_tw_stop_visible_all(session, &upd_select->tw);
     else
-        return (!__wt_txn_tw_start_visible_all(session, &upd_select->tw));
+        visible_all = __wt_txn_tw_start_visible_all(session, &upd_select->tw);
+
+    if (visible_all)
+        return (false);
+
+    /*
```

Comment:
Here is an alternate comment suggestion if that helps:

```
/*
 * Update chains are only saved if there are newer uncommitted updates or for the in-memory storage.
 * If there has not been an on-disk value before, a new key has been inserted. We need to save updates
 * if there is at least one update that has not been aborted.
 */
```

Reply:
Thanks. Adjusted the comment to understand the need for saving the update chain.